connect/src/main/java/com/stripe/android/connect/webview/StripeConnectWebViewContainer.kt

```diff
@@ -0,0 +1,289 @@
+package com.stripe.android.connect.webview
+
+import android.annotation.SuppressLint
+import android.graphics.Bitmap
+import android.view.LayoutInflater
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.FrameLayout
+import androidx.annotation.RestrictTo
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.doOnAttach
+import androidx.core.view.doOnDetach
+import androidx.core.view.isVisible
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import com.stripe.android.connect.BuildConfig
+import com.stripe.android.connect.EmbeddedComponentManager
+import com.stripe.android.connect.PrivateBetaConnectSDK
+import com.stripe.android.connect.StripeEmbeddedComponent
+import com.stripe.android.connect.appearance.Appearance
+import com.stripe.android.connect.databinding.StripeConnectWebviewBinding
+import com.stripe.android.connect.webview.serialization.AccountSessionClaimedMessage
+import com.stripe.android.connect.webview.serialization.ConnectInstanceJs
+import com.stripe.android.connect.webview.serialization.PageLoadMessage
+import com.stripe.android.connect.webview.serialization.SecureWebViewMessage
+import com.stripe.android.connect.webview.serialization.SetterMessage
+import com.stripe.android.connect.webview.serialization.toJs
+import com.stripe.android.core.Logger
+import com.stripe.android.core.version.StripeSdkVersion
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.jsonObject
+
+@PrivateBetaConnectSDK
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface StripeConnectWebViewContainer {
+    /**
+     * Set the [EmbeddedComponentManager] to use for this view.
+     * Must be called when this view is created via XML.
+     * Cannot be called more than once per instance.
+     */
+    fun setEmbeddedComponentManager(embeddedComponentManager: EmbeddedComponentManager)
+}
+
+@OptIn(PrivateBetaConnectSDK::class)
+internal interface StripeConnectWebViewContainerInternal {
+    /**
+     * Update the appearance of the Connect instance.
+     */
+    fun updateConnectInstance(appearance: Appearance)
+
+    /**
+     * Load the given URL in the WebView.
+     */
+    fun loadUrl(url: String)
+}
+
+@OptIn(PrivateBetaConnectSDK::class)
+internal class StripeConnectWebViewContainerImpl(
```

Comment:
👀 Code here should resemble code in `PayoutsView` previously. It should contain "dumb" view code whereas the Controller holds the domain logic we want to test more extensively